app/frontend/src/features/communities/CommunityPage/CommunityPage.tsx

```diff
@@ -0,0 +1,233 @@
+      width: "30%",
+    },
+  },
+  cardContainer: {
+    [theme.breakpoints.down("xs")]: {
+      //break out of page padding
+      width: "100vw",
+      position: "relative",
+      left: "50%",
+      right: "50%",
+      marginLeft: "-50vw",
+      marginRight: "-50vw",
+    },
+    [theme.breakpoints.up("sm")]: {
+      marginTop: theme.spacing(1),
+      marginBottom: theme.spacing(2),
+      display: "flex",
+      flexDirection: "row",
+      flexWrap: "wrap",
+      justifyContent: "space-between",
+      "&::after": {
+        content: "''",
+        flexBasis: "100%",
+        [theme.breakpoints.up("sm")]: {
+          flexBasis: `calc(50% - ${theme.spacing(1)})`,
+        },
+        [theme.breakpoints.up("md")]: {
+          flexBasis: `calc(33.33% - ${theme.spacing(1)})`,
+        },
+      },
+    },
+  },
+  loadMoreButton: {
+    alignSelf: "center",
+    [theme.breakpoints.up("sm")]: {
+      width: `calc(50% - ${theme.spacing(1)})`,
+    },
+    [theme.breakpoints.up("md")]: {
+      width: `calc(33% - ${theme.spacing(1)})`,
+    },
+  },
+  placeEventCard: {
+    marginBottom: theme.spacing(1),
+    width: 200,
+    [theme.breakpoints.up("sm")]: {
+      width: `calc(50% - ${theme.spacing(1)})`,
+    },
+    [theme.breakpoints.up("md")]: {
+      width: `calc(33% - ${theme.spacing(1)})`,
+    },
+  },
+}));
+
+export default function CommunityPage() {
+  const classes = useCommunityPageStyles();
+
+  const { communityId, communitySlug } = useParams<{
+    communityId: string;
+    communitySlug?: string;
+  }>();
+
+  const {
+    isLoading: isCommunityLoading,
+    error: communityError,
+    data: community,
+  } = useCommunity(+communityId);
+
+  const history = useHistory();
+  useEffect(() => {
+    if (!community) return;
+    if (community.slug !== communitySlug) {
+      // if the address is wrong, redirect to the right place
+      history.replace(routeToCommunity(community.communityId, community.slug));
+    }
+  }, [community, communitySlug, history]);
+
+  if (!communityId)
+    return <Alert severity="error">Invalid community id.</Alert>;
+
+  if (isCommunityLoading)
+    return <CircularProgress className={classes.center} />;
+
+  if (!community || communityError)
+    return (
+      <Alert severity="error">
+        {communityError?.message || "Error loading the community."}
+      </Alert>
+    );
+
+  return (
+    <div className={classes.root}>
+      <HeaderImage community={community} className={classes.header} />
+      <div className={classes.topContainer}>
+        <div className={classes.topInfo}>
+          <Breadcrumbs aria-label="breadcrumb" className={classes.breadcrumbs}>
+            {community.parentsList
+              .filter((parent) => !!parent.community)
+              .map((parent) => (
+                <Link
+                  to={routeToCommunity(
+                    parent.community!.communityId,
+                    parent.community!.slug
+                  )}
```

Comment:
```suggestion
              .filter((parent) : parent is ParentWithCommunity => !!parent.community)
              .map((parent) => (
                <Link
                  to={routeToCommunity(
                    parent.community.communityId,
                    parent.community.slug
                  )}
```
where `ParentWithCommunity` is this interface extending `Parent.AsObject`: `{ community: CommunityParent.AsObject; }` seems to "correct" the type enough so you don't need non-null assertion